### PR TITLE
use HTTP API in vscode plugin

### PR DESCRIPTION
**Pull request by alexflint**
This pr changes the vscode plugin to post events over HTTP. It removes:
- all use of unix domain sockets
- all diff suggestions logic

### CODE REVIEW

File: vscode/Makefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DEST := "~/.vscode/extensions/kite.vscode"
```
Comment (jlozano) [inline vscode/Makefile]: Are we only using VSCode on osx/linux?

File: vscode/README.md [hunk 2]
```diff
@@ -1,24 +1,21 @@
-# KiteCode
+# VSCode plugin for Kite
 
-KiteCode is a plugin that integrates Kite, the programming assistant into Visual Studio code. Kite will automagically suggest library usages, point out syntactic errors, and will generally work with you to make (Python) development less painful.
+KiteCode is a plugin that integrates Kite, the programming assistant into
```
Comment (jlozano) [inline vscode/README.md]: I think you need a comma after assistant

File: vscode/README.md [hunk 3]
```diff
@@ -1,24 +1,21 @@
...
-Run ```npm install``` in the base directory of this repository.
-
-## Testing
-
-Open Visual Studio Code, navigate to this project directory, and change the launch target from `Launch Extension` to `Launch Tests`. Then, hit the green play triangle. Testing from the command line to come as soon as I can figure it out.
-
-## Support
-
-Kite is Mac OS only, and this extension makes use of UNIX sockets, so this extension only works on Mac OS.
+Then start vscode and you should see Kite in the list of extensions.
```
Comment (jlozano) [inline vscode/README.md]: `Kite` -> `KiteCode`

File: vscode/extension.js [hunk 4]
```diff
@@ -94,29 +90,43 @@ var KiteOutgoing = class {
...
+      port: '46624',
+      path: endpoint,
+      method: 'POST',
     };
 
-    let msg = JSON.stringify(event);
-    this.outgoingSocket.send(msg, 0, msg.length, KiteOutgoing.PORT, KiteOutgoing.HOST);
+    var req = http.request(options);
+    req.write(payload);
+    req.end();
```
Comment (jlozano) [inline vscode/extension.js]: are we guaranteed that write will finish before we call `req.end` ?

File: vscode/Makefile (hunk 1 above)
Comment (alexflint) [inline vscode/Makefile]: actually no, but this is just for the dev workflow, and I do believe that at least on linux the path is the same (and windows doesn't even have `make` by default`)

File: vscode/extension.js (hunk 4 above)
Comment (alexflint) [inline vscode/extension.js]: I believe... I took this from a nodejs example of sending an http request

File: vscode/README.md (hunk 2 above)
Comment (alexflint) [inline vscode/README.md]: OH DAMN!

Comment (alexflint) [inline vscode/README.md]: : )

File: vscode/Makefile (hunk 1 above)
Comment (jlozano) [inline vscode/Makefile]: lol of course it doesn't, honestly do not know why I assumed it did:)

File: vscode/extension.js (hunk 4 above)
Comment (jlozano) [inline vscode/extension.js]: hmm i think you are correct http://blog.mixu.net/2011/02/01/understanding-the-node-js-event-loop/

**Pull request merged**